public/js/reducers/sources.js

```diff
@@ -81,13 +79,19 @@ function update(state = initialState, action) {
   return state;
 }
 
-function getFilenameFromUrl(sourceUrl) {
+function _updateSource(source) {
+  const pathname = getPathnameFromUrl(source.url);
+  const filename = pathname.substring(pathname.lastIndexOf("/") + 1);
+  return Object.assign({}, source, { pathname, filename });
+}
+
+function getPathnameFromUrl(sourceUrl) {
   if (!sourceUrl) {
     return "";
   }
 
   const url = new URL(sourceUrl);
```

Comment:
Yea, that makes a lot of sense. We could do a simple feature check `if (URL) {} else {}`